starknet_py/contract.py

```diff
@@ -172,14 +181,40 @@ async def call(
         )
         return self._payload_transformer.to_python(result)
 
-    async def invoke(self, signature: Optional[Collection[int]] = None) -> InvokeResult:
+    async def invoke(
+        self,
+        signature: Optional[Collection[int]] = None,
+        max_fee: Optional[int] = None,
+        auto_estimate: bool = False,
+    ) -> InvokeResult:
         """
         Invokes a method.
 
         :param signature: Signature to send
+        :param max_fee: Max amount of Wei to be paid when executing transaction
+        :param auto_estimate: Use automatic fee estimation, not recommend as it may lead to high costs
         :return: InvokeResult
         """
-        tx = self._make_invoke_function(signature)
+        if auto_estimate and max_fee is not None:
+            raise ValueError(
+                "Max_fee and auto_estimate are exclusive and cannot be provided at the same time."
+            )
+        if auto_estimate and self.max_fee is not None:
+            raise ValueError(
+                "Auto_estimate cannot be used if max_fee was provided when preparing a function call."
+            )
```

Comment:
Logically yes, they can be merged, since `self.max_fee` is usually set when doing a `.prepare()` so I though having a separate error message would be beneficial.